akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -295,11 +295,24 @@ private[http] object HttpServerBluePrint {
 
     override def onUpstreamFailure(error: Throwable, ctx: Context[ResponseRenderingContext]) =
       error match {
+        case EntityStreamSizeException(limit, contentLength) ⇒
+          val status = StatusCodes.RequestEntityTooLarge
+          val summary = contentLength match {
+            case Some(cl) ⇒ s"Request Content-Length of $cl bytes exceeds the configured limit of $limit bytes"
+            case None     ⇒ s"Aggregated data length of request entity exceeds the configured limit of $limit bytes"
+          }
+          val info = ErrorInfo(summary, "Consider increasing the value of akka.http.server.parsing.max-content-length")
```

Comment:
Yeah, I guess it's OK as the actual applied limit is part of the summary. Folks will find out that `akka.http.server.parsing.max-content-length` isn't what they're looking for when it's different from `limit`.